vpc/tool/container2/setup_container_linux.go

```diff
@@ -259,144 +258,137 @@ func addIPv6AddressAndRoutes(ctx context.Context, allocation types.Allocation, n
-	}
-	err := nsHandle.AddrAdd(link, &new6Addr)
-	if err != nil {
-		logger.G(ctx).WithError(err).Error("Unable to add IPv6 addr to link")
-		return errors.Wrap(err, "Unable to add IPv6 addr to link")
-	}
-
-	var linkLocalAddr netlink.Addr
-	for {
-		// We need *a* permanent address
-		addrs, err := nsHandle.AddrList(link, netlink.FAMILY_V6)
-		if err != nil {
-			return fmt.Errorf("Cannot get addrs: %w", err)
-		}
-
-		for _, addr := range addrs {
-			if addr.Scope == unix.RT_SCOPE_LINK {
-				linkLocalAddr = addr
-				goto out
-			}
-		}
-		select {
-		case <-ctx.Done():
-			return fmt.Errorf("Did not find permanent link-local address in set: %v: %w", addrs, ctx.Err())
-		case <-time.After(100 * time.Millisecond):
-		}
-	}
-
-out:
 	origns, err := netns.Get()
-	defer origns.Close()
 	if err != nil {
 		return fmt.Errorf("Could not get current net ns: %w", err)
 	}
+	defer origns.Close()
 
 	err = netns.Set(netns.NsHandle(netnsfd))
 	if err != nil {
 		return fmt.Errorf("Could not switch net ns: %w", err)
 	}
 
-	networkInterface, err := net.InterfaceByName("eth0")
-	if err != nil {
-		return fmt.Errorf("Could not get network interface eth0: %w", err)
+	var mtu *uint32
+	var routes []struct {
+		mtu   uint32
+		route net.IPNet
 	}
+	for _, route := range allocation.Routes {
+		if route.Family != vpcapi.AssignIPResponseV3_Route_IPv6 {
+			continue
+		}
 
-	conn, _, err := ndp.Dial(networkInterface, ndp.Addr(linkLocalAddr.IP.String()))
-	if err != nil {
-		return fmt.Errorf("Unable to dial NDP conn: %w", err)
+		_, routeNet, err := net.ParseCIDR(route.Destination)
+		if err != nil {
+			logger.G(ctx).WithError(err).Error("Could not parse route")
+			return fmt.Errorf("Could not parse route CIDR (%s): %w", route.Destination, err)
+		}
+
+		ones, zeros := routeNet.Mask.Size()
+		if routeNet.IP.Equal(net.IPv6zero) && ones == 0 && zeros == 128 {
+			mtu = &route.Mtu
+		} else {
+			routes = append(routes, struct {
+				mtu   uint32
+				route net.IPNet
+			}{mtu: route.Mtu, route: *routeNet})
+		}
 	}
-	defer conn.Close()
 
-	var filter ipv6.ICMPFilter
-	filter.SetAll(true)
-	filter.Accept(ipv6.ICMPTypeRouterAdvertisement)
-	err = conn.SetICMPFilter(&filter)
+	acceptRAPrefixInfo, err := os.OpenFile("/proc/sys/net/ipv6/conf/eth0/accept_ra_pinfo", os.O_RDWR, 0)
 	if err != nil {
-		return fmt.Errorf("Unable to setup ICMP filter %v: %w", filter, err)
+		return fmt.Errorf("Could not open accept_ra_pinfo: %w", err)
 	}
-
-	m := &ndp.RouterSolicitation{
-		Options: []ndp.Option{
-			&ndp.LinkLayerAddress{
-				Direction: ndp.Source,
-				Addr:      networkInterface.HardwareAddr,
-			},
-		},
+	_, err = acceptRAPrefixInfo.WriteString("0")
+	_ = acceptRAPrefixInfo.Close()
+	if err != nil {
+		return fmt.Errorf("Could not write 0 to accept_ra_pinfo: %w", err)
 	}
 
-	/* Poll for the router */
-	var routerAddr net.IP
-	for {
-		err = ctx.Err()
-		if err != nil {
-			return fmt.Errorf("Context complete before NDP finished: %w", err)
+	err = func() error {
```

Comment:
why this anon function? 

Why not a standalone function called `setAdvertisedMTU` ?